File: 049. Group Anagrams/Python/Solution.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        memory = {}
        for word in strs:
            sorted_word = ''.join(sorted(word))
            if sorted_word not in memory:
                memory[sorted_word] = []
            memory[sorted_word].append(word)
        return list(memory.values())

File: 049. Group Anagrams/C++/Solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string> &strs) {
        unordered_map<string, vector<string>> memory;
        for (const auto &word : strs) {
            string sortedWord = word;
            sort(sortedWord.begin(), sortedWord.end());
            memory[sortedWord].push_back(word);
        }
        vector<vector<string>> result;
        for (const auto &item : memory) {
            result.push_back(item.second);
        }
        return result;
    }
};